### ISSUE TITLE
Use the "live" NETCoreApp shared framework in CoreCLR and Libraries tests

### ISSUE THREAD

**Issue by dagood** [dotnet/runtime#487]
Now that the repo is consolidated, we can look at ways to reduce some redundant infrastructure around the shared framework.

Before consolidation, Core-Setup was downstream from CoreCLR and CoreFX. This means the crossgenned NETCoreApp shared framework produced by Core-Setup doesn't go through CoreCLR and CoreFX's tests.

CoreCLR tests do their own crossgen to set up a R2R testing infrastructure. I think CoreFX does something similar, but I'm not sure if it crossgens.

With consolidation, we could have CoreCLR and Libraries tests use the R2R shared framework prepared by Installer. This might save build time, and seems like it will remove some redundant infra so we don't have to maintain it.

The current Installer sharedfx generation infra might not be clean enough yet for this to work in a repo that has much higher dev workflow standards (IMO). I have concerns around incrementality and efficiency in a tight dev loop. I'm not active on that part of the infra quite enough to pinpoint the problems offhand, but I want to make sure we consider it before going all-in.

---

(I thought I remembered idea/plan in an issue or a roadmap, but I can't find it. Filing to make sure it's tracked regardless.)

/cc @trylek @stephentoub

**Comment by trylek** [comment]
Alluding to Davis' comment - CoreCLR way of crossgenning the framework is totally hacky too. My initial strawman was to switch this over to the SuperIlc tool I originally wrote for the purpose of local Crossgen2 testing - as one of its side effects it's capable of crossgenning framework (using both the legacy and new Crossgen) employing parallelization so it's quite a bit faster. As an additional bonus it would give us at least some code coverage for the tool too.

**Comment by jkotas** [comment]
>  current Installer sharedfx generation infra 

I agree that it would make sense to use the installer sharedfx generation infra to produce the bits that the tests runs against.

> SuperIlc tool

SuperIlc is a test harness. I do not think it makes sense to use it to build the product.

**Comment by jkotas** [comment]
(Issues should be marked with a single area only: https://github.com/dotnet/runtime/blob/master/docs/issues-pr-management.md#scenarios-where-we-all-have-to-work-together)

**Comment by trylek** [comment]
Well, OK, I think that my point is - on the one hand we have super hacky code in

https://github.com/dotnet/runtime/blob/a76c402d16499ab2f05953edbdb92abfa68357b9/src/coreclr/build-test.cmd#L653

or another copy in

https://github.com/dotnet/runtime/blob/a76c402d16499ab2f05953edbdb92abfa68357b9/src/coreclr/build-test.sh#L145

or another copy in the installer pipeline. What would be your preferred way to get rid of this wasteful redundancy?

[Sorry about the issue tags.]

**Comment by jkotas** [comment]
Have just the one in the installer pipeline.

**Comment by safern** [comment]
Something to consider here is to run a set of libraries tests in the single PR pipeline when the installer subset is changed. Look at discussion: https://github.com/dotnet/runtime/pull/1473#discussion_r367146918

**Comment by ViktorHofer** [comment]
Closing as done.

**Comment by ViktorHofer** [comment]
Reopening as I misread the title.

**Comment by ViktorHofer** [comment]
When ever we get to this we should make use of the shared framework's SDK functionality to dump the bundle layout onto disk and use that for both the runtime and libraries.

Moving to 7.0.
